### PR TITLE
refactor(types): enable noImplicitany in transaction.ts

### PR DESCRIPTION
**Pull request by joecervino**
Fixes #<issue_number_goes_here> (it's a good idea to open an issue first for discussion)

- [ ] Tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: src/codec.ts [hunk 1]
```diff
@@ -250,7 +250,7 @@ function decode(value: Value, type: s.Type): Value {
       decoded = new Date(decoded);
       break;
     case s.TypeCode.ARRAY:
-      decoded = decoded.map(value => {
+      decoded = decoded.map((value: Value) => {
```
Comment (AVaksman) [inline src/codec.ts]: why do you need type here?
decoded is type `Value` = `any`

File: src/transaction.ts [hunk 2]
```diff
@@ -20,77 +20,82 @@ import {EventEmitter} from 'events';
...
-
-export interface KeyRange {
-  startClosed?: Value[];
-  startOpen?: Value[];
-  endClosed?: Value[];
-  endOpen?: Value[];
+export interface Fields {
+  [key: string]: spanner_client.protobuf.IValue;
+}
+export interface KeyRange extends spanner_client.spanner.v1.IKeyRange {
```
Comment (AVaksman) [inline src/transaction.ts]: where are you using this interface?

File: src/codec.ts (hunk 1 above)
Comment (joecervino) [reply src/codec.ts]: Don't need it, but didn't think it'd hurt unless this is a no go? (`Value` has type `any` 🙌 )

File: src/transaction.ts (hunk 2 above)
Comment (joecervino) [reply src/transaction.ts]: On line 933: `const fields: Fields = {};`

**Pull request closed without merging**